### PR TITLE
Add eye, zeros, zeros_like, ones, ones_like

### PR DESCRIPTION
**Pull request by jcrist**
Add implementations of `eye`, `zeros`, and `zeros_like`, mirroring the equivalent numpy api.

### CODE REVIEW

**Comment by mrocklin** [comment]
Looks like docstrings need to be spaced slightly differently.  Maybe a difference in Numpy version?

**Comment by jcrist** [comment]
Grumble. Since `NORMALIZE_WHITESPACE` only treats whitespace variations as equivalent (and not equivalent with the empty string), there's no way to AFAICT to get get the doctests to work with multiple versions of numpy if they differ in output between

```
np.array([ 0.0, 0.0])  # and
np.array([0.0, 0.0])
```

Seems to be only an issue with float arrays, just skipped those tests.

**Comment by rgommers** [comment]
The normal approach is to only run doctests for the latest version of numpy. With all the print changes around numpy 1.14, that's the only reasonable thing to do in CI.

File: sparse/coo/common.py [hunk 1]
```diff
@@ -705,3 +705,122 @@ def roll(a, shift, axis=None):
...
+        data_length = max(min(data_length, N + k), 0)
+        m_coords = np.arange(data_length, dtype=np.intp)
+        n_coords = m_coords - k
+    else:
+        n_coords = m_coords = np.arange(data_length, dtype=np.intp)
+
+    coords = np.stack([n_coords, m_coords])
+    data = np.ones(data_length, dtype=dtype)
+
+    return COO(coords, data=data, shape=(N, M), has_duplicates=False)
```
Comment (hameerabbasi) [inline sparse/coo/common.py]: Might be nice to add `sorted=True`.

File: sparse/coo/common.py [hunk 2]
```diff
@@ -705,3 +705,122 @@ def roll(a, shift, axis=None):
...
+    array([[0, 0],
+           [0, 0]])
+    """
+    from .core import COO
+
+    if not isinstance(shape, tuple):
+        shape = (shape,)
+    data = np.empty(0, dtype=dtype)
+    coords = np.empty((len(shape), 0), dtype=np.intp)
+    return COO(coords, data=data, shape=shape, has_duplicates=False)
```
Comment (hameerabbasi) [inline sparse/coo/common.py]: `sorted=True`

File: sparse/coo/common.py [hunk 3]
```diff
@@ -705,3 +705,122 @@ def roll(a, shift, axis=None):
...
+    out : COO
+        Array of zeros with the same shape and type as `a`.
+
+    Examples
+    --------
+    >>> x = np.ones((2, 3), dtype='i8')
+    >>> zeros_like(x).todense()  # doctest: +NORMALIZE_WHITESPACE
+    array([[0, 0, 0],
+           [0, 0, 0]])
+    """
```
Comment (hameerabbasi) [inline sparse/coo/common.py]: `sorted=True`.

File: sparse/coo/common.py [hunk 4]
```diff
@@ -705,3 +705,122 @@ def roll(a, shift, axis=None):
...
+        m_coords = n_coords + k
+    elif k < 0:
+        data_length = max(min(data_length, N + k), 0)
+        m_coords = np.arange(data_length, dtype=np.intp)
+        n_coords = m_coords - k
+    else:
+        n_coords = m_coords = np.arange(data_length, dtype=np.intp)
+
+    coords = np.stack([n_coords, m_coords])
+    data = np.ones(data_length, dtype=dtype)
```
Comment (hameerabbasi) [inline sparse/coo/common.py]: Data as a scalar should work here. Takes less memory due to scalar broadcasting being a view.

**Comment by mrocklin** [comment]
That's what we do in dask/dask.  This is probably Hameer's call though.

**Comment by jcrist** [comment]
Thanks for the review, I believe all comments have been addressed.

**Comment by hameerabbasi** [comment]
Thabks for making these changes! They’re very welcome. Merging now.

**Pull request merged**